src/scripts/main.js

```diff
@@ -1,3 +1,172 @@
 'use strict';
 
 // write code here
+document.addEventListener('DOMContentLoaded', function () {
+  const employeeTable = document.querySelector('table');
+  let currentSort = { column: null, order: 'asc' };
+  let selectedRow = null;
+
+  const headers = document.querySelectorAll('th');
+
+  headers.forEach((header, index) => {
+    header.addEventListener('click', () => {
+      const order =
+        currentSort.column === index && currentSort.order === 'asc'
+          ? 'desc'
+          : 'asc';
+
+      sortTable(employeeTable, index, order);
+      currentSort = { column: index, order: order };
+    });
+  });
+
+  function sortTable(table, columnIndex, order) {
+    const rows = Array.from(table.tBodies[0].rows);
+    const isNumeric = columnIndex === 3 || columnIndex === 4;
+
+    rows.sort((rowA, rowB) => {
+      let a = rowA.cells[columnIndex].innerText.trim();
+      let b = rowB.cells[columnIndex].innerText.trim();
+
+      if (isNumeric) {
+        a = parseFloat(a.replace(/[^0-9.-]+/g, ''));
+        b = parseFloat(b.replace(/[^0-9.-]+/g, ''));
+      }
+
+      if (order === 'asc') {
+        return isNumeric ? a - b : a.localeCompare(b);
+      } else {
+        return isNumeric ? b - a : b.localeCompare(a);
+      }
+    });
+
+    rows.forEach((row) => table.tBodies[0].appendChild(row));
```

Comment:
Appending rows one by one to the table body causes unnecessary reflows and can degrade performance. A better approach would be to create a DocumentFragment, append the rows to that, and then append the DocumentFragment to the table body. This way, the browser only has to update the DOM once.